### PR TITLE
EDP renewal change

### PR DESCRIPTION
**Pull request by Atemo**
* **Addressed Issue(s)**: #3484

* **Server Mode**: renewal

* **Description of Pull Request**: 
Switch to a different weapon or unequip a weapon now cancel EDP
EDP can't be used with fist (right hand check only)

### CODE REVIEW

File: conf/battle/skill.conf [hunk 1]
```diff
@@ -375,3 +375,7 @@ banana_bomb_duration: 0
 // Official: no
 // Legacy rAthena logic: yes
 skill_drop_items_full: no
+
+// Switch/unequip right hand weapon should remove EDP and EDP cannot be used with fist? (Note 1)
```
Comment (cydh) [inline conf/battle/skill.conf]: `fist` -> `bare hand`?

File: src/map/battle.cpp [hunk 2]
```diff
@@ -8522,6 +8522,7 @@ static const struct _battle_data {
 	{ "autoloot_adjust",                    &battle_config.autoloot_adjust,                 0,      0,      1,              },
 	{ "broadcast_hide_name",                &battle_config.broadcast_hide_name,             2,      0,      NAME_LENGTH,    },
 	{ "skill_drop_items_full",              &battle_config.skill_drop_items_full,           0,      0,      1,              },
+	{ "switch_remove_edp",                  &battle_config.switch_remove_edp,               0,      0,      1,              },
```
Comment (cydh) [inline src/map/battle.cpp]: default to yes, so `1, 0, 1`

**Comment by Atemo** [comment]
I was wondering if the setting
```
// EDP setting: (Note 3)
// On official renewal server switch/unequip right hand weapon should remove EDP and EDP can't be used with bare hand.
// 0: Disabled (rAthena legacy and pre-renewal behaviour).
// 1: Enable the setting on pre-renewal compilation.
// 2: Enable on renewal compilation.
// 3: 1+2
switch_remove_edp: 2
```
is better. Disabled by default on pre-renewal compilation, enabled by default on renewal

File: conf/battle/skill.conf [hunk 3]
```diff
@@ -375,3 +375,11 @@ banana_bomb_duration: 0
 // Official: no
 // Legacy rAthena logic: yes
 skill_drop_items_full: no
+
+// EDP setting: (Note 3)
+// On official renewal server switch/unequip right hand weapon should remove EDP and EDP can't be used with bare hand.
+// 0: Disabled (rAthena legacy and pre-renewal behaviour).
+// 1: Enable the setting on pre-renewal compilation.
+// 2: Enable on renewal compilation.
+// 3: 1+2
```
Comment (aleos89) [inline conf/battle/skill.conf]:
```
// EDP setting: (Note 3)
// When switching/unequipping a right hand weapon, should EDP be removed? EDP can't be used with bare hand.
// 0: Disabled (rAthena legacy and pre-renewal behavior).
// 1: Enabled on pre-renewal.
// 2: Enabled on renewal.
// 3: 1+2
```

**Comment by Daegaladh** [comment]
I think it should have a 4th option, disabled on pre-renewal + enabled in renewal, and be the default one, so it have the official behaviour for the 2 modes by default.

**Pull request merged**